strongbox-web-core/src/main/java/org/carlspring/strongbox/app/boot/BootProgressBeanPostProcessor.java

```diff
@@ -0,0 +1,192 @@
+    }
+
+    @Override
+    public void setApplicationContext(ApplicationContext applicationContext)
+    {
+        this.applicationContext = applicationContext;
+    }
+
+    @Override
+    public void setBeanName(String beanName)
+    {
+        this.beanName = beanName;
+    }
+
+    @EventListener
+    public void onApplicationEvent(ContextRefreshedEvent event)
+    {
+        DefaultListableBeanFactory factory = (DefaultListableBeanFactory) applicationContext.getAutowireCapableBeanFactory();
+        factory.destroySingleton(beanName);
+    }
+
+    @PostConstruct
+    public void startup() {
+        progress = BehaviorSubject.create();
+    }
+
+    @PreDestroy
+    public void destroy()
+            throws InterruptedException
+    {
+        logger.debug("Notifying clients Strongbox has booted.");
+        progress.onComplete();
+
+        // Slightly delay stopping the server to leave enough time for existing responses to complete.
+        // Fixes
+        //   curl: (18) transfer closed with outstanding read data remaining
+        //   err_incomplete_chunked_encoding
+        Thread.sleep(1600);
+
+        logger.debug("Stopping server {}", webServer.toString());
+        webServer.stop();
+        webServer = null;
+
+        progress = null;
+
+        logger.debug("Removing registered {} event listeners", BootProgressBeanPostProcessor.class);
+        Collection<ApplicationListener<?>> listeners = ((AbstractApplicationContext) applicationContext).getApplicationListeners();
+
+        ApplicationListener<?> applicationListener = listeners.stream()
+                                                              .filter(listener -> listener instanceof ApplicationListenerMethodAdapter &&
+                                                                              ((ApplicationListenerMethodAdapter) listener)
+                                                                                      .toString()
+                                                                                      .contains(BootProgressBeanPostProcessor.class.getSimpleName()))
+                                                              .findFirst()
+                                                              .get();
+
+        ApplicationEventMulticaster applicationEventMulticaster = applicationContext.getBean(
+                AbstractApplicationContext.APPLICATION_EVENT_MULTICASTER_BEAN_NAME, ApplicationEventMulticaster.class);
+        applicationEventMulticaster.removeApplicationListener(applicationListener);
+    }
+
+    @Nullable
+    public Object postProcessBeforeInitialization(Object bean, String beanName)
+            throws BeansException
+    {
+        logger.trace("Found bean: {}", beanName);
+
+        if(displayMessages.containsKey(beanName) || beanName.contains("org.springframework.security.authentication.DefaultAuthenticationEventPublisher"))
```

Comment:
`DefaultAuthenticationEventPublisher.class.getName()` would be better than `"org.springframework.security.authentication.DefaultAuthenticationEventPublisher"`. What if `spring-security-core` moves it to another package or drop it ?

Reply:
Good point! I've also replaced some of the classes in the `displayMessages` as well (wherever possible)